patch update scripts to work with debian package together

Index: roundcube/bin/update.sh
===================================================================
--- roundcube.orig/bin/update.sh	2016-05-05 11:01:51.998496673 +0200
+++ roundcube/bin/update.sh	2016-05-05 11:04:44.663488854 +0200
@@ -18,6 +18,13 @@
  | Author: Thomas Bruederli <roundcube@gmail.com>                        |
  +-----------------------------------------------------------------------+
 */
+if ($_SERVER['RCMAIL_CONFIG_DIR']) {
+  define('RCMAIL_CONFIG_DIR', $_SERVER['RCMAIL_CONFIG_DIR']);
+}
+
+if ($_SERVER['DEBIAN_PKG']) {
+  define('DEBIAN_PKG', TRUE);
+}
 
 define('INSTALL_PATH', realpath(__DIR__ . '/..') . '/' );
 
@@ -86,20 +93,26 @@
       if ($opts['accept'] || strtolower($input) == 'y') {
         $error = $written = false;
 
-        // backup current config
-        echo ". backing up the current config file(s)...\n";
-
-        foreach (array('config', 'main', 'db') as $file) {
-          if (file_exists(RCMAIL_CONFIG_DIR . '/' . $file . '.inc.php')) {
-            if (!copy(RCMAIL_CONFIG_DIR . '/' . $file . '.inc.php', RCMAIL_CONFIG_DIR . '/' . $file . '.old.php')) {
-              $error = true;
+        if (!DEBIAN_PKG) {
+          // backup current config
+          echo ". backing up the current config file(s)...\n";
+
+          foreach (array('config', 'main', 'db') as $file) {
+            if (file_exists(RCMAIL_CONFIG_DIR . '/' . $file . '.inc.php')) {
+              if (!copy(RCMAIL_CONFIG_DIR . '/' . $file . '.inc.php', RCMAIL_CONFIG_DIR . '/' . $file . '.old.php')) {
+                $error = true;
+              }
             }
           }
         }
 
         if (!$error) {
           $RCI->merge_config();
-          echo ". writing " . RCMAIL_CONFIG_DIR . "/config.inc.php...\n";
+          if (DEBIAN_PKG) {
+            echo ". writing " . RCMAIL_CONFIG_DIR . "/config.inc.php.dpkg-new...\n";
+          } else {
+            echo ". writing " . RCMAIL_CONFIG_DIR . "/config.inc.php...\n";
+          }
           $written = $RCI->save_configfile($RCI->create_config());
         }
 
@@ -114,9 +127,11 @@
               echo "- '" . $msg['prop'] . ($msg['name'] ? "': " . $msg['name'] : "'") . "\n";
           }
 
-          if ($RCI->legacy_config) {
-            foreach (array('main', 'db') as $file) {
-              @unlink(RCMAIL_CONFIG_DIR . '/' . $file . '.inc.php');
+          if (!DEBIAN_PKG) {
+            if ($RCI->legacy_config) {
+              foreach (array('main', 'db') as $file) {
+                @unlink(RCMAIL_CONFIG_DIR . '/' . $file . '.inc.php');
+              }
             }
           }
         }
@@ -153,11 +168,13 @@
     echo "Please check the 'mime_types' config option and run this script again.\n";
   }
 
-  // check database schema
-  if ($RCI->config['db_dsnw']) {
-    echo "Executing database schema update.\n";
-    $success = rcmail_utils::db_update(INSTALL_PATH . 'SQL', 'roundcube', $opts['version'],
-        array('errors' => true));
+  if (!DEBIAN_PKG) {
+    // check database schema
+    if ($RCI->config['db_dsnw']) {
+      echo "Executing database schema update.\n";
+      $success = rcmail_utils::db_update(INSTALL_PATH . 'SQL', 'roundcube', $opts['version'],
+          array('errors' => true));
+    }
   }
 
   // update composer dependencies
Index: roundcube/program/include/iniset.php
===================================================================
--- roundcube.orig/program/include/iniset.php	2016-05-05 11:01:51.998496673 +0200
+++ roundcube/program/include/iniset.php	2016-05-05 11:01:51.994496612 +0200
@@ -36,6 +36,10 @@
     define('RCUBE_LOCALIZATION_DIR', INSTALL_PATH . 'program/localization/');
 }
 
+if (!defined('DEBIAN_PKG')) {
+    define('DEBIAN_PKG', FALSE);
+}
+
 define('RCUBE_INSTALL_PATH', INSTALL_PATH);
 define('RCUBE_CONFIG_DIR',  RCMAIL_CONFIG_DIR.'/');
 
Index: roundcube/program/include/rcmail_install.php
===================================================================
--- roundcube.orig/program/include/rcmail_install.php	2016-05-05 11:01:51.998496673 +0200
+++ roundcube/program/include/rcmail_install.php	2016-05-05 11:13:22.863400563 +0200
@@ -104,10 +104,22 @@
                 $this->legacy_config = true;
             }
 
+            // Debian save its old config in dpkg-bak file
+            if (DEBIAN_PKG && $config = $this->load_config_file(RCUBE_CONFIG_DIR . 'main.inc.php.dpkg-bak')) {
+                $this->config        = array_merge($this->config, $config);
+                $this->legacy_config = true;
+            }
+
             if ($config = $this->load_config_file(RCUBE_CONFIG_DIR . 'db.inc.php')) {
                 $this->config        = array_merge($this->config, $config);
                 $this->legacy_config = true;
             }
+
+            // Debian save its old config in dpkg-bak file
+            if (DEBIAN_PKG && $config = $this->load_config_file(RCUBE_CONFIG_DIR . 'db.inc.php.dpkg-bak')) {
+                $this->config        = array_merge($this->config, $config);
+                $this->legacy_config = true;
+            }
         }
 
         $this->configured = !empty($config);
@@ -267,8 +279,10 @@
             }
 
             // save change
-            $this->config[$prop] = $value;
-            $config[$prop]       = $value;
+            if (!DEBIAN_PKG || $prop != 'db_dsnw') {
+                $this->config[$prop] = $value;
+                $config[$prop]       = $value;
+            }
         }
 
         $out = "<?php\n\n";
@@ -280,6 +294,11 @@
             $out .= "\$config['$prop'] = " . self::_dump_var($value, $prop) . ";\n\n";
         }
 
+        if (DEBIAN_PKG) {
+            $out .= "/* Do not set db_dsnw here, use dpkg-reconfigure roundcube-core to configure database ! */\n";
+            $out .= "include_once(\"/etc/roundcube/debian-db-roundcube.php\");\n";
+        }
+
         return $out;
     }
 
@@ -290,8 +309,14 @@
      */
     public function save_configfile($config)
     {
+        $filename = "config.inc.php";
+
+        if (DEBIAN_PKG) {
+            $filename = "config.inc.php.dpkg-new";
+        }
+
         if (is_writable(RCUBE_CONFIG_DIR)) {
-            return file_put_contents(RCUBE_CONFIG_DIR . 'config.inc.php', $config);
+            return file_put_contents(RCUBE_CONFIG_DIR . $filename, $config);
         }
 
         return false;
